tests/pages/ViewportPageObject.ts

```diff
@@ -0,0 +1,192 @@
+};
+
+export interface IViewportPageObject {
+  doubleClickOn: (point: { x: number; y: number }) => Promise<void>;
+  clicksOn: (
+    points: { x: number; y: number }[],
+    button?: 'left' | 'right' | 'middle'
+  ) => Promise<void>;
+  normalizedClicksOn: (
+    normalizedPoints: { x: number; y: number }[],
+    button?: 'left' | 'right' | 'middle'
+  ) => Promise<void>;
+  normalizedDragOn: (params: NormalizedDragParams) => Promise<void>;
+  orientationMarkers: {
+    topMid: Locator;
+    leftMid: Locator;
+    rightMid: Locator;
+    bottomMid: Locator;
+  };
+  overlayText: {
+    topLeft: Locator;
+    topRight: Locator;
+    bottomLeft: Locator;
+    bottomRight: Locator;
+  };
+  overlayMenu: {
+    dataOverlay: {
+      button: Locator;
+      click: () => Promise<void>;
+    };
+    orientation: {
+      button: Locator;
+      click: () => Promise<void>;
+    };
+    windowLevel: {
+      button: Locator;
+      click: () => Promise<void>;
+    };
+  };
+  pane: Locator;
+  svg: (innerElement?: SvgInnerElement) => Locator;
+}
+
+export class ViewportPageObject {
+  readonly page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  private getOrientationMarkers(viewport: Locator) {
+    return {
+      topMid: viewport.locator('.top-mid.orientation-marker'),
+      leftMid: viewport.locator('.left-mid.orientation-marker'),
+      rightMid: viewport.locator('.right-mid.orientation-marker'),
+      bottomMid: viewport.locator('.bottom-mid.orientation-marker'),
+    };
+  }
+
+  private getOverlayText(viewport: Locator) {
+    return {
+      topLeft: viewport.getByTestId('viewport-overlay-top-left'),
+      topRight: viewport.getByTestId('viewport-overlay-top-right'),
+      bottomLeft: viewport.getByTestId('viewport-overlay-bottom-left'),
+      bottomRight: viewport.getByTestId('viewport-overlay-bottom-right'),
+    };
+  }
+
+  private getOverlayMenu(viewport: Locator) {
+    return {
+      get dataOverlay() {
+        const button = viewport.locator('[data-cy^="dataOverlayMenu"]').first();
+        return {
+          button,
+          async click() {
+            await button.click();
+          },
+        };
+      },
+      get orientation() {
+        const button = viewport.locator('[data-cy^="orientationMenu"]');
+        return {
+          button,
+          async click() {
+            await button.click();
+          },
+        };
+      },
+      get windowLevel() {
+        const button = viewport.locator('[data-cy^="windowLevelMenu"]');
+        return {
+          button,
+          async click() {
+            await button.click();
+          },
+        };
+      },
+    };
+  }
+
+  private getSvg(viewport: Locator, innerElement?: SvgInnerElement) {
+    return viewport.locator(`svg.svg-layer${innerElement ? ` ${innerElement}` : ''}`);
+  }
+
+  public viewportPageObjectFactory(viewport: Locator): IViewportPageObject {
```

Comment:
I don't think this should be public.

Reply:
I thought on you maybe using your custom locator to get a viewport element, and them you'd be able to get all the features of the `ViewportPageObject` on top of this custom locator. But given the YAGNI rule I'll move it to private for now.